rxandroidble/src/main/java/com/polidea/rxandroidble/internal/serialization/ConnectionOperationQueueImpl.java

```diff
@@ -16,25 +21,33 @@
 import rx.functions.Action1;
 import rx.functions.Cancellable;
 
-public class ConnectionOperationQueueImpl implements ConnectionOperationQueue {
+@ConnectionScope
+public class ConnectionOperationQueueImpl implements ConnectionOperationQueue, ConnectionSubscriptionWatcher {
 
-    private OperationPriorityFifoBlockingQueue queue = new OperationPriorityFifoBlockingQueue();
+    private final String deviceMacAddress;
+    private final DisconnectionRouterOutput disconnectionRouterOutput;
+    private Subscription disconnectionThrowableSubscription;
+    private final OperationPriorityFifoBlockingQueue queue = new OperationPriorityFifoBlockingQueue();
     private final Future<?> runnableFuture;
     private volatile boolean shouldRun = true;
-    private volatile BleDisconnectedException bleDisconnectedException = null;
+    private volatile BleException disconnectionException = null;
```

Comment:
Effectively `volatile` is not required here because of synchronization. Having synchronized access guarantees that the other thread will read the value *after* the previous has written it.